core/src/main/java/org/cru/contentscoring/core/servlets/ResourceUrlMapperServlet.java

```diff
@@ -0,0 +1,77 @@
+package org.cru.contentscoring.core.servlets;
+
+import com.day.cq.commons.Externalizer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.felix.scr.annotations.sling.SlingServlet;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.request.RequestParameter;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * This resource url mapper servlet is used to determine the external URL(s) of a given resource
+ * based on the publisher's sling mapping.
+ */
+@SlingServlet(
+    methods = "GET",
+    paths = "/bin/cru/url/mapper"
+)
+public class ResourceUrlMapperServlet extends SlingSafeMethodsServlet {
```

Comment:
Every time we write a new "/bin/..." servlet I cringe a little bit. I wish we could use jax-rs resource classes instead. I asked Nestor about setting this up one time, and he thought it would be possible, but we never took it beyond that.
Do you think it'd be worth our time to look into that? Not for this PR, probably.

Reply:
Here's one solution that has arisen in that space: [caravan-jaxrs](https://github.com/wcm-io-caravan/caravan-jaxrs)